### PR TITLE
[#5509] Allow cryptography 3.2 with OpenSSL 1.0.2.

### PR DESCRIPTION
**Pull request by dumol**
Scope
=====

Allow cryptography 3.2 on platforms with OpenSSL 1.0.2.

This is an issue on `python-package`, where the necessary `CRYPTOGRAPHY_ALLOW_OPENSSL_102` env var is set for normal `compat` tests, but not propagated for elevated tests run through `sudo`.

Changes
=======

Export the relevant env var unconditionally.

**Drive-by change**:
  * updated `pip` to latest version.
  * small fix for AIX
  * support for OpenBSD 6.8 and FreeBSD 12.

How to try and test the changes
===============================

reviewers: @adiroiban 

Please review changes.

Check that `compat` tests ran from this branch are green with `cryptography` 3.2.1 on Amazon 2 and CentOS 7 (check the patch diff for the branch toggle):

https://chevah.com/buildbot/builders/python-package-amzn-2/builds/95
https://chevah.com/buildbot/builders/python-package-rhel-7/builds/166

### CODE REVIEW

File: chevah/compat/__init__.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
     DefaultAvatar = UnixDefaultAvatar
     SuperAvatar = UnixSuperAvatar
 
+    # Unconditionally allow cryptography 3.2.1 with OpenSSL 1.0.2.
+    os.environ['CRYPTOGRAPHY_ALLOW_OPENSSL_102']  = 'yes'
```
Comment (adiroiban) [inline chevah/compat/__init__.py:34]: I guess that lint will complain about this:

```suggestion
    os.environ['CRYPTOGRAPHY_ALLOW_OPENSSL_102'] = 'yes'
```

Comment (dumol) [reply chevah/compat/__init__.py]: Yes, thanks, I have already fixed that locally.

**Comment by dumol** [comment]
**needs-review**

**Review by adiroiban — APPROVED** [review]
Thanks.

You can release a new version by compat by running

```
$ ./build-compat/bin/python setup.py publish
```
assuming that your VPN is a active and that your `~/.pypirc` file contains the info for Chevah PyPI server.

File: release-notes.rst [hunk 2]
```diff
@@ -2,6 +2,12 @@ Release notes for chevah.compat
 ===============================
 
 
+0.58.3 - 2020-08-05
```
Comment (adiroiban) [inline release-notes.rst:5]:
```suggestion
0.58.3 - 2020-11-09
```

**Comment by adiroiban** [comment]
I have released 0.58.3  to Chevah PyPI server.

**Comment by adiroiban** [comment]
I have merged it since it was released. Thanks!

**Comment by dumol** [comment]
> I have released 0.58.3 to Chevah PyPI server.

Thanks! I was waiting for the results of `compat` tests for all `python-package` slaves yesterday evening.

For the record, all current `python-package` slaves have passed the compat tests in this branch: https://chevah.com/buildbot/builders/python-package-group-all/builds/39.

```
trigger triggered python-package-alpine-312 python-package-amzn-2 python-package-obsd python-package-rhel-7 python-package-rhel-8 python-package-ubuntu-1804 python-package-ubuntu-2004 python-package-win-2016 python-package-win-2019 python-package-aix-71-dev python-package-macos-1013 python-package-rhel-5 python-package-aix-53 python-package-fbsd-12 python-package-hpux python-package-sol-10 python-package-sol-10-sparc python-package-sol-112 python-package-sol-112-sparc python-package-sol-11 python-package-ubuntu-1604-arm64 ( 2 hrs, 19 mins, 24 secs )
- no logs -
python-package-hpux #52 SUCCESS
python-package-sol-10 #87 SUCCESS
python-package-aix-53 #58 SUCCESS
python-package-sol-11 #80 SUCCESS
python-package-amzn-2 #98 SUCCESS
python-package-fbsd-12 #11 SUCCESS
python-package-aix-71-dev #79 SUCCESS
python-package-rhel-7 #169 SUCCESS
python-package-sol-112-sparc #22 SUCCESS
python-package-ubuntu-1804 #153 SUCCESS
python-package-ubuntu-1604-arm64 #61 SUCCESS
python-package-sol-10-sparc #23 SUCCESS
python-package-obsd #151 SUCCESS
python-package-ubuntu-2004 #64 SUCCESS
python-package-alpine-312 #31 SUCCESS
python-package-rhel-8 #93 SUCCESS
python-package-macos-1013 #139 SUCCESS
python-package-win-2019 #262 SUCCESS
python-package-sol-112 #24 SUCCESS
python-package-win-2016 #89 SUCCESS
python-package-rhel-5 #12 SUCCESS
```

**Pull request merged**